app/notification.go

```diff
@@ -160,19 +160,35 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 	mentionedUsersList := make([]string, 0, len(mentions.Mentions))
 	updateMentionChans := []chan *model.AppError{}
 
+	// for each mention, make sure to update thread autofollow
+	for id := range mentions.Mentions {
+		umc := make(chan *model.AppError, 1)
+		go func(userId string) {
+			defer close(umc)
+			if *a.Config().ExperimentalSettings.ThreadAutoFollow && post.RootId != "" {
+				nErr := a.Srv().Store.Thread().CreateMembershipIfNeeded(userId, post.RootId)
+				if nErr != nil {
+					umc <- model.NewAppError("SendNotifications", "app.channel.autofollow.app_error", nil, nErr.Error(), http.StatusInternalServerError)
+					return
+				}
+			}
+			umc <- nil
+		}(id)
+		updateMentionChans = append(updateMentionChans, umc)
```

Comment:
This is going to create log warnings that was unable to update the channel mention count, that is not related to this. I think we should have another list channels here, to make it independent and generate the logs in the right way.

Reply:
@jespino done